### PR TITLE
Budget-app

### PR DESCRIPTION
**Pull request by ibrahim777764**
###                                                                  Budget-app

1. Used ruby gems as software packages system.
2. Installed Ruby on Rails framework.
3. Understood Rails RESTful design and router.
4. Used controllers to handle requests and render empty views.
5. Used params from browser request in a safe way.
6. Used preprocessed html file with embedded Ruby code.
7. Used layouts and templates for shared content.
8. Used database migration files to maintain database schema.
9. Used validations for models.
10. Secured app from n+1 problems.
11. Understood what ORM is.
12. Wrote SQL queries with ActiveRecord.
13. Set up associations between models.
14. Builded a webapp that requires the user to log in.
15. Used devise gem for authentication.
16. Limit access to webapp resources based on authorization rules.
17. Analyzed in writing why you have made a coding choice using one structure over another.

video [presentation]( https://www.loom.com/share/19bf2269c10649b1a02084c0f7bb51ac)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,105 @@
...
+- Ruby
+- Ruby on Rails
+- PostgreSql
+- TailwindCSS
 
-* Database creation
+## Getting Started
 
-* Database initialization
+ [live demo](https://tranquil-sierra-36809.herokuapp.com/)
```
Comment (AnselemOdims) [inline README.md]: - Please kindly ensure that the live version actually works because at the moment it does not.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,105 @@
...
+5. Open a Pull Request
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Thanks to Microverse
+
```
Comment (AnselemOdims) [inline README.md]: - [ ] Please kindly include an acknowledgment that gives appropriate credit to the original design owner

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -5,7 +5,7 @@
 		<p class="fw-bold">$<%= @group.total_amount.round(2) %></p>
   </div>
   <div class="p-4 pt-0">
-    <%=image_tag @group.icon, class: 'img-thumbnail ratio ratio-21x9' %>
+    
     <p class="text-muted text-end pt-2"><%= @group.name %></p>
   </div>
 </div>
```
Comment (AnselemOdims) [inline app/views/groups/show.html.erb:11]: - [ ] Please kindly ensure that the login and signup buttons on the splash screen matches the original design

### Your Design
image

### Original Design
image

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @ibrahim777764 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [x] Please as stated in the requirements, kindly record a presentation video(preferrably using Loom) of no more than 5mins and include the link in the README
- [x] I realized that after signing up and logging out. When I try to log back back it keeps taking me back to the signup page. Please help fix this.
- [x] Please kindly include a back button in the `transaction`, `add new category`, and `add new transaction` pages for seamless navigation back and forth.
- [x] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 4]
```diff
@@ -1,24 +1,110 @@
...
+- Ruby
+- Ruby on Rails
+- PostgreSql
+- TailwindCSS
 
-* Database creation
+## Getting Started
 
-* Database initialization
+ [live demo](https://secure-wave-18374.herokuapp.com/)
```
Comment (RolandM99) [inline README.md]: - [ ] Kindly, make sure that you're using the correct link for the live demo of your project. For now, you're using the wrong link which is not updated with the new features you implemented after the first review. Please, fix this issue.

**Review by RolandM99 — CHANGES_REQUESTED** [review]
### ♻️  REQUSTED CHANGES 

Hi @ibrahim777764 , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 5]
```diff
@@ -1,24 +1,110 @@
...
```
Comment (EricMbouwe) [inline README.md]: - [ ] Let's set up a working deployment of your app and update this link.

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @ibrahim777764, 

Great work on making the changes requested by a previous reviewer 👏 🔥
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights 🎉🔥
- No linter errors ✔
- Descriptive PR ✔

 # Required Changes ♻️
- [ ]  After considering your message in the submitting form, what is allowed is if you can add the last requirement on Saturday no problem, Saturday night is the deadline. 

_Check the comments under the review._

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ibrahim777764** [comment]
I submitted again, but I didn’t change any thing
That a same work. . Because now I don’t have electricity and I can’t do anything. And  my laptop charging is finished, and  i  submitted the same project again, so i just want to be approved. Because I already completed whole project, the only issue with deplying tha branch which I can’t do before Saturday night because of electricity problem . And  i just want to be approved , i don't wanna repeat a module because of  this small problem,please help me on this 
thankyou

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @ibrahim777764👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**